src/transition.rs

```diff
@@ -74,10 +74,29 @@ impl TransitionKind {
                     project
                 };
 
-                return (model.to_string(), new_project.to_string());
+                return Ok((model.to_string(), new_project.to_string()));
+            }
+        }
+
+        // Fallback in case transition is not defined
+        match self {
+            TransitionKind::Automatic => {
+                // Keep current firmware variant
+                Ok((model.to_string(), project.to_string()))
+            },
+            TransitionKind::Open => if project == "76ec" {
+                // Allow current firmware variant, it is already open
+                Ok((model.to_string(), project.to_string()))
+            } else {
+                Err(format!("Model '{}' is not supported by open firmware and EC at this time", model))
+            }
+            TransitionKind::Proprietary => if project != "76ec" {
+                // Allow current firmware variant, it is already proprietary
+                Ok((model.to_string(), project.to_string()))
+            } else {
+                Err(format!("Model '{}' is not supported by proprietary firmware and EC at this time", model))
```

Comment:
It might be neater to condense this to something with less redundancy and nesting:

```rust
match self {
    TransitionKind::Open if project != "76ec" => {
        Err(format!("Model '{}' is not supported by open firmware and EC at this time", model))
    }
    TransitionKind::Proprietary if project == "76ec" => {
        Err(format!("Model '{}' is not supported by proprietary firmware and EC at this time", model))
    }
    _ => Ok((model.to_string(), project.to_string()))
}
```

But it should behave the same. Everything here looks sensible otherwise.

Reply:
Yeah, that's cleaner, I will convert